nixos/modules/services/backup/postgresql-backup.nix

```diff
@@ -7,28 +7,43 @@ let
   cfg = config.services.postgresqlBackup;
 
   postgresqlBackupService = db: dumpCmd:
-    {
+    let
+      suffix = getAttr cfg.compression {
+        "none" = "";
+        "gzip" = ".gz";
+        "zstd" = ".zstd";
+      };
+      compressCmd = getAttr cfg.compression {
+        "none" = "cat";
+        "gzip" = "${pkgs.gzip}/bin/gzip -c";
+        "zstd" = "${pkgs.zstd}/bin/zstd -c";
+      };
+      curFile = "${cfg.location}/${db}.sql${suffix}";
+      prevFile = "${cfg.location}/${db}.prev.sql${suffix}";
+      inProgressFile = "${cfg.location}/${db}.in-progress.sql${suffix}";
+    in {
       enable = true;
 
       description = "Backup of ${db} database(s)";
 
       requires = [ "postgresql.service" ];
 
-      path = [ pkgs.coreutils pkgs.gzip config.services.postgresql.package ];
+      path = [ pkgs.coreutils config.services.postgresql.package ];
 
       script = ''
         set -e -o pipefail
 
         umask 0077 # ensure backup is only readable by postgres user
 
-        if [ -e ${cfg.location}/${db}.sql.gz ]; then
-          mv ${cfg.location}/${db}.sql.gz ${cfg.location}/${db}.prev.sql.gz
+        if [ -e ${curFile} ]; then
+          mv ${curFile} ${prevFile}
```

Comment:
This doesn't replace the old previous file if `compression` has changed since the last run, which is a space leak and a privacy liability.

Reply:
Updated it to remove all possible previous files before moving.